internal/executor/cache.go

```diff
@@ -61,14 +64,52 @@ func DownloadCache(executor *Executor, commandName string, cacheHost string, ins
 		folderToCache = filepath.Join(custom_env["CIRRUS_WORKING_DIR"], folderToCache)
 	}
 
-	cachePopulated, cacheAvailable := tryToDownloadAndPopulateCache(logUploader, commandName, cacheHost, cacheKey, folderToCache)
+	var glob string
+	baseFolder := folderToCache
+	foldersToCache := []string{folderToCache}
+
+	if pathLooksLikeGlob(folderToCache) {
+		glob = folderToCache
+
+		// Sanity check
+		//
+		// When glob is used, the semantics are clearly defined only when
+		// it's matches are scoped to the current working directory,
+		// because otherwise it's impossible to make paths inside of the
+		// archive portable (i.e. independent on the location of the working
+		// directory).
+		//
+		// Note: this is not a security stop-gap but merely a hint to the users
+		// that they are doing something potentially wrong.
+		terminatedWorkingDir := custom_env["CIRRUS_WORKING_DIR"]
+
+		if !strings.HasSuffix(terminatedWorkingDir, string(os.PathSeparator)) {
+			terminatedWorkingDir += string(os.PathSeparator)
+		}
 
-	var folderToCacheHash = ""
-	var folderToCacheFileHashes = make(map[string]string)
-	if cachePopulated {
-		folderToCacheHash, folderToCacheFileHashes, err = hasher.FolderHash(folderToCache)
+		if !strings.HasPrefix(glob, terminatedWorkingDir) {
```

Comment:
Will it be easier to disallow use of `..` or use `filepath.Abs` that resolves `..` and then check if the resulting path starts with the working directory?

Reply:
Can you provide a use-case when this will be a better approach?

Reply:
It seems `../../**/passwords` will pass the check? I think you can add `glob = filepath.Abs(folderToCache)` to fix it. 🤔

Reply:
It don't think it'll pass the check since non-absolute `folderToCache`'s are joined with `custom_env["CIRRUS_WORKING_DIR"]`.

Reply:
Oh, I see 🤦‍♂️

I've thought about using `filepath.Abs()`, but I was unsure if it'll actually work fine on non-existent paths which cache folder globs may produce before the cache is populated.

What do you think about using `filepath.Abs()` here instead:

https://github.com/cirruslabs/cirrus-ci-agent/blob/bb4d0755b038c67b1850e4ed417a0a8674cab0d1/internal/executor/cache.go#L60-L62

?

Reply:
Sounds good. Can user it there 👌

Reply:
Implemented in https://github.com/cirruslabs/cirrus-ci-agent/pull/54/commits/879d4fa65f017f5fcb4657314b8fc7a17fec63fd.